exercises/binary-search/.meta/generator/binary_search_case.rb

```diff
@@ -0,0 +1,25 @@
+require 'generator/exercise_case'
+
+class BinarySearchCase < Generator::ExerciseCase
+  def workload
+    [binary_search, assertion].join
+  end
+
+  private
+
+  def binary_search
+    "binary = BinarySearch.new(#{array})\n"
+  end
+
+  def assertion
+    expected.is_a?(Fixnum) ? standard_assertion : error_assertion
+  end
+
+  def standard_assertion
+    "assert_equal #{expected}, binary.search_for(#{value})"
+  end
+
+  def error_assertion
+    assert_raises(RuntimeError, "binary.search_for(#{value})")
```

Comment:
We may not want to raise an error... the specifications state:

> "comments": [
      "The error object is used to indicate that the value is not included in the array.",
      "It should be replaced with the respective expression that is idiomatic",
      "for the language that implements the tests.",

Reply:
What would you recommend? I checked a few other implementations and [some return -1](https://github.com/exercism/javascript/blob/master/exercises/binary-search/example.js) and other [return value exception](https://github.com/exercism/python/blob/master/exercises/binary-search/example.py). 

`ArgumentError` might make sense, but then one could argue that the argument itself is valid.

Reply:
I think we should return what makes sense for Ruby.

Given the description, I think the failure point is giving a collection that is not sorted, we should raise an error, as an argument error.  This should be the only error that we draw out for the tests.

Given the description in the Readme there is no indication of a failure for finding no value:

> If the remaining array to be searched is empty, then the key cannot be found in the array and a special "not found" indication is returned.

That special "not found" indication is probably `nil` for Ruby.   This fits in with idiomatic Ruby.  We get something truthy or falsey, the value found, or `nil`.  Further, this is expected "normal operation" rather than "exceptional behavior".  We shouldn't raise an exception for this, in my opinion.

Reply:
Agreed that value not found is not "exceptional behavior". I'll make the necessary changes to return `nil`.